File: articles/moral-injury-bureaucracy-and-distributed-responsibility/go/moral_injury_score.go
// Synthetic moral injury risk example.
// Educational scaffold only.
//
// Run:
//   go run go/moral_injury_score.go

package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func main() {
	conscienceConflict := 0.9
	institutionalBetrayal := 0.8
	distributedResponsibility := 0.7
	exposure := 0.8
	scarcity := 0.6
	accountability := 0.4
	protectedVoice := 0.3
	repairCapacity := 0.3

	latent := 0.30*conscienceConflict +
		0.30*institutionalBetrayal +
		0.25*distributedResponsibility +
		0.30*exposure +
		0.20*scarcity -
		0.25*accountability -
		0.20*protectedVoice -
		0.20*repairCapacity

	risk := logistic(latent)

	fmt.Printf("Synthetic moral injury risk: %.3f\n", risk)
}

File: articles/moral-injury-bureaucracy-and-distributed-responsibility/c/moral_injury_score.c
/*
Synthetic moral injury risk example.
Educational scaffold only.

Compile:
  cc c/moral_injury_score.c -lm -o outputs/tables/moral_injury_score_c
Run:
  ./outputs/tables/moral_injury_score_c
*/

#include <math.h>
#include <stdio.h>

double logistic(double x) {
    return 1.0 / (1.0 + exp(-x));
}

int main(void) {
    double conscience_conflict = 0.9;
    double institutional_betrayal = 0.8;
    double distributed_responsibility = 0.7;
    double exposure = 0.8;
    double scarcity = 0.6;
    double accountability = 0.4;
    double protected_voice = 0.3;
    double repair_capacity = 0.3;

    double latent =
        0.30 * conscience_conflict +
        0.30 * institutional_betrayal +
        0.25 * distributed_responsibility +
        0.30 * exposure +
        0.20 * scarcity -
        0.25 * accountability -
        0.20 * protected_voice -
        0.20 * repair_capacity;

    double risk = logistic(latent);

    printf("Synthetic moral injury risk: %.3f\n", risk);
    return 0;
}
